[BUG] Rolling std/var test fails on `aarch64`

**Describe the bug**
On aarch64 device, `test_rolling.py::test_rolling_var_std_large` fails.

**Steps/Code to reproduce bug**
Compile and run tests on an `aarch64` device.

**Expected behavior**
No test failure

**Environment overview (please complete the following information)**
 - Environment location: Docker
 - Method of cuDF install: Source

**Environment details**
- Arm64 AWS instance with T4G running docker image `gpuci/rapidsai-arm64:21.10-cuda11.2-devel-ubuntu18.04-py3.7`
- Building branch `21.10` from source inside the docker

**Additional context**
<details>
<summary>Failure message</summary>

```
E   AssertionError: DataFrame.iloc[:, 0] (column name="0") are different
E   
E   DataFrame.iloc[:, 0] (column name="0") values are different (0.1 %)
E   [index]: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, ...]
E   [left]:  [nan, 4002012973754642.0, 1.621628019214138e+18, 1.5055229335517686e+18, 1.2130050452685005e+18, 9.14100107310882e+16, 5791637470694680.0, nan, nan, 1.4844868785972754e+18, 2.1645853041147346e+18, 2.3375835157311446e+17, 3.934563365182552e+18, nan, nan, 1.1690167197593986e+17, 4.1773673896015936e+17, 8182561312342576.0, 2.136436259893138e+16, 9.032267068739712e+17, 3937329286284264.5, nan, nan, nan, nan, 1.027981782236456e+18, 1.318117582978819e+18, nan, nan, nan, 4.3670581272924805e+18, nan, nan, nan, nan, nan, 1002347030958050.0, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, 3.6350638726874726e+17, nan, nan, nan, 6.060113894365609e+17, 4.179663285856369e+17, 5428099376715536.0, 2.661934221691759e+18, 3.444024360021037e+17, 1.635838474447556e+16, 3.311671721863526e+17, 2.6780583904167373e+18, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, 1.1268988510768224e+18, nan, nan, nan, nan, nan, nan, nan, 5.328411990059997e+17, nan, nan, 7.660893689835269e+17, nan, nan, 4.076597413244933e+17, 9.212365061956675e+17, 4.005646604888652e+18, 3.709992034100594e+16, 9608592810066220.0, nan, nan, nan, nan, nan, nan, 24659543895012.5, nan, nan, 6.089884131899407e+18, 4.53878424129967e+17, nan, ...]
E   [right]: [nan, 4002012973754642.0, 1.621628019214138e+18, 1.5055229335517686e+18, 1.2130050452685005e+18, 9.14100107310882e+16, 5791637470694688.0, nan, nan, 1.4844868785972754e+18, 2.1645853041147346e+18, 2.3375835157311437e+17, 3.934563365182552e+18, nan, nan, 1.1690167197593986e+17, 4.1773673896015936e+17, 8182561312342560.0, 2.1364362598931364e+16, 9.032267068739712e+17, 3937329286284264.5, nan, nan, nan, nan, 1.027981782236456e+18, 1.318117582978819e+18, nan, nan, nan, 4.3670581272924805e+18, nan, nan, nan, nan, nan, 1002347030958050.0, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, 3.6350638726874726e+17, nan, nan, nan, 6.060113894365609e+17, 4.179663285856369e+17, 5428099376715528.0, 2.661934221691759e+18, 3.444024360021034e+17, 1.6358384744475312e+16, 3.311671721863523e+17, 2.678058390416737e+18, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, 1.1268988510768224e+18, nan, nan, nan, nan, nan, nan, nan, 5.328411990059997e+17, nan, nan, 7.660893689835269e+17, nan, nan, 4.076597413244933e+17, 9.212365061956675e+17, 4.005646604888652e+18, 3.709992034100577e+16, 9608592810066048.0, nan, nan, nan, nan, nan, nan, 24659543895012.5, nan, nan, 6.089884131899407e+18, 4.538784241299675e+17, nan, ...]
```
<details/>

--- a/python/cudf/cudf/tests/test_rolling.py
+++ b/python/cudf/cudf/tests/test_rolling.py
@@ -136,7 +136,7 @@ def test_rolling_with_offset(agg):
 @pytest.mark.parametrize("agg", ["std", "var"])
 @pytest.mark.parametrize("ddof", [0, 1])
 @pytest.mark.parametrize("center", [True, False])
-@pytest.mark.parametrize("seed", [42, 100])
+@pytest.mark.parametrize("seed", [100, 2000])
 @pytest.mark.parametrize("window_size", [2, 10, 100])
 def test_rolling_var_std_large(agg, ddof, center, seed, window_size):
     if PANDAS_GE_110:
@@ -151,27 +151,26 @@ def test_rolling_var_std_large(agg, ddof, center, seed, window_size):
     flower_bound = -math.sqrt(abs(np.finfo(np.float64).min) / window_size)
 
     n_rows = 1_000
-    cardinality = 100
     data = rand_dataframe(
         dtypes_meta=[
             {
                 "dtype": "int64",
                 "null_frequency": 0.4,
-                "cardinality": cardinality,
+                "cardinality": n_rows,
                 "min_bound": ilower_bound,
                 "max_bound": iupper_bound,
             },
             {
                 "dtype": "float64",
                 "null_frequency": 0.4,
-                "cardinality": cardinality,
+                "cardinality": n_rows,
                 "min_bound": flower_bound,
                 "max_bound": fupper_bound,
             },
             {
                 "dtype": "decimal64",
                 "null_frequency": 0.4,
-                "cardinality": cardinality,
+                "cardinality": n_rows,
                 "min_bound": ilower_bound,
                 "max_bound": iupper_bound,
             },
@@ -186,13 +185,18 @@ def test_rolling_var_std_large(agg, ddof, center, seed, window_size):
     expect = getattr(pdf.rolling(window_size, 1, center), agg)(ddof=ddof)
     got = getattr(gdf.rolling(window_size, 1, center), agg)(ddof=ddof)
 
-    # Due to pandas-37051, pandas rolling var/std on uniform window is
-    # not reliable. Skipping these rows when comparing.
-    for col in expect:
-        mask = (got[col].fillna(-1) != 0).to_pandas()
-        expect[col] = expect[col][mask]
-        got[col] = got[col][mask]
-        assert_eq(expect[col], got[col], **kwargs)
+    import platform
+
+    if platform.machine() == "aarch64":
+        # Due to pandas-37051, pandas rolling var/std on uniform window is
+        # not reliable. Skipping these rows when comparing.
+        for col in expect:
+            mask = (got[col].fillna(-1) != 0).to_pandas()
+            expect[col] = expect[col][mask]
+            got[col] = got[col][mask]
+            assert_eq(expect[col], got[col], **kwargs)
+    else:
+        assert_eq(expect, got, **kwargs)
 
 
 @pytest.mark.xfail

